app/document-management/js/views/advanced_search.js

```diff
@@ -105,7 +105,7 @@ define([
         onSubmitForm: function () {
             var queryString = this.constructQueryString();
             if (queryString) {
-                App.router.navigate(encodeURIComponent(App.config.workspaceId) + '/search/' + encodeURIComponent(queryString), {trigger: true});
+	            App.router.navigate(encodeURIComponent(App.config.workspaceId) + '/search/' + encodeURIComponent(queryString), {trigger: true});
```

Comment:
Without encodeURIComponent on the whole query string (in addition of applying on each parameter) , when they are included in searched values, 3 reserved characters (i.e #, & and +) are decoded by backbone router (router.js line 101, query parameter) prior to sending request. So the searched strings are modified. 
Examples :
- if we search documents with attribute dieze=machin#machine
request url become : http://localhost:8281/eplmp-server-rest/api/workspaces/My-wks/documents/search?&attributes=TEXT:dieze:machin
in this case, in addition of decoding, hashtag # is removed (jquery.js line 7843)
- if we search documents with attribute etcom=et&com
request url : http://localhost:8281/eplmp-server-rest/api/workspaces/My-wks/documents/search?&attributes=TEXT:etcom:et&com&from=0&size=10000
in this case, & is considered as a parameter separator in server side.
- if we search documents with attribute plus=pl+us
request url : http://localhost:8281/eplmp-server-rest/api/workspaces/My-wks/documents/search?&attributes=TEXT:plus:pl+us&from=0&size=10000
in this case, + is decoded as espace in server side, so the value of attribute become pl us instead of pl+us.

So, to avoid such behavior, we must apply encoding on the whole query string after having encoded each component of query. In this case, special characters are preserved after the first decoding by router.